src/ln/channelmanager.rs

```diff
@@ -1939,7 +1939,16 @@ impl ChannelManager {
 					return Err(MsgHandleErrInternal::send_err_msg_no_close("Got a message for a channel from the wrong node!", msg.channel_id));
 				}
 				if !chan.is_usable() {
-					return Err(MsgHandleErrInternal::from_no_close(HandleError{err: "Channel not yet available for receiving HTLCs", action: Some(msgs::ErrorAction::IgnoreError)}));
+					// If the update_add is completely bogus, the channel will reject it outright,
+					// but if we've sent a shutdown but they haven't acknowledged it yet, we just
```

Comment:
nit: maybe instead of they, the "forward peer" and "we reject the new HTLC and so fail backward"?